lib/wanda/execution.ex

```diff
@@ -58,12 +58,14 @@ defmodule Wanda.Execution do
       end
     end)
     |> Enum.any?(fn name ->
-      %{group_id: g} =
+      group_id ==
         name
         |> :global.whereis_name()
-        |> :sys.get_state()
-
-      g == group_id
+        |> get_group_id()
```

Comment:
This is unfortunately a bit "unsafe" since the GenServer could be down (e.g. it concurrently processed a message that causes its graceful stop or termination, like the latest receive facts message) and this could cause an exception.
Also, the server could be finding a slow database causing this call to hang, as it processes one message at a time.
That's why, even if this is better than the state introspection, I wouldn't go with this solution.

Reply:
Ok man! Even though the risk to happen such a scenario is small, it is there, so let's avoid it. I already implemented the other version. I will update the contracts first.